pkg/providers/aws/provider_postgres_metrics.go

```diff
@@ -0,0 +1,203 @@
+// postgres metric provider scrapes metrics for a single postgres (rds) instance
+//
+// we are required to gather data from postgres (rds) instances which can be used in valuable
+// alerts, to ensure and monitor performance of postgres (rds) instances
+//
+// this providers does
+//   * scrape metric data from cloudwatch
+//   * build a generic cloud metric data type from cloudwatch data
+//   * return generic cloud metric data to metric controller to be exposed
+//
+// this provider does not
+//   * expose the metrics, this is controller at a higher level (controller)
+//
+package aws
+
+import (
+	"context"
+	"fmt"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/cloudwatch"
+	"github.com/aws/aws-sdk-go/service/cloudwatch/cloudwatchiface"
+	"github.com/integr8ly/cloud-resource-operator/pkg/apis/integreatly/v1alpha1"
+	"github.com/integr8ly/cloud-resource-operator/pkg/providers"
+	"github.com/integr8ly/cloud-resource-operator/pkg/resources"
+	errorUtil "github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"strings"
+	"time"
+)
+
+const (
+	postgresMetricProviderName = "aws postgres metrics provider"
+	cloudWatchRDSDBDimension   = "DBInstanceIdentifier"
+)
+
+// builds a list of the metric types and the corresponding metric statistics we want
+var rdsCloudWatchMetricTypes = []providers.CloudProviderMetricType{
+	{
+		PromethuesMetricName: "cro_postgres_free_storage_average",
+		ProviderMetricName:   "FreeStorageSpace",
+		Statistic:            "Average",
+	},
+	{
+		PromethuesMetricName: "cro_postgres_free_storage_average_maximum",
+		ProviderMetricName:   "FreeStorageSpace",
+		Statistic:            "Maximum",
+	},
+	{
+		PromethuesMetricName: "cro_postgres_free_storage_average_minimum",
+		ProviderMetricName:   "FreeStorageSpace",
+		Statistic:            "Minimum",
+	},
```

Comment:
What's the reasoning behind the min and max statistics? Is there an additional cost for this?

Reply:
After discussion remove the min and the max.